docroot/themes/humsci/su_humsci_theme/js/su_humsci_theme.js

```diff
@@ -46,7 +46,13 @@
         });
       });
 
-      $("img.lazy", context).lazyload();
+      $('img.lazy', context).each(function () {
+        if ($(this).closest('.masonry').length) {
```

Comment:
So this is to find the images in a masonry group and then set their src and load them right away because all images get the loading image at first. Yeah, I guess you have to do this with assuming all images are 'lazy'.

Reply:
yeah that seemed to be a quick fix. the library i'm using doesn't provide callbacks when an image is loaded to rearrange the masonry. and the thumbnails i found don't stretch themselves to the `width` and `height` attributes. the loading.png only stretches its width but not height.

Reply:
Another alternative that might work is to use more jquery. set inline styles that set the height with the correct aspect ratio of the attributes, but takes the actual width into consideration.
like an image: `<img src=".." width="800" height="1200" />` but its container is only 250px wide. the jquery would set inline styles to `style="width: 250px; height: 375px"`. but then it would need to adjust that on responsive. :man_shrugging:

Reply:
You would have to add the https://imagesloaded.desandro.com/ package.

Lots of people use masonry and that package together but your solution seems light enough. You don't get the lazy load in masonry tho. Might be worth looking in to that library.

Reply:
thats what is being used in the masonry. https://github.com/SU-HSDO/suhumsci/blob/develop/docroot/themes/humsci/su_humsci_theme/patterns/organisms/masonry/masonry.ui_patterns.yml#L37
but the lazy loading library i'm using doesnt have a callback to trigger the `.layout()` method

Reply:
Gotcha.